### ISSUE TITLE
Block reader not updating when block changes

### ISSUE THREAD

**Issue by RNADNA47** [CyclopsMC/IntegratedDynamics#538]
#### Issue type:
- :bug: Bug

____
#### Short description:

Pictured is a block reader atop a block. Stone is placed in that spot, an AA atomic reconstructor with the lens of the minor changes the block to ore, then the block is broken. The variables essentially detect the block, and when block == stone a redstone signal is sent to deactivate the auto-breaker.

The pictured block reader's block value is displayed in the panel. For some reason, the block reader isn't reading the current value of the block. Breaking the cable temporarily fixes the reader, but the issue returns on the next block change.

#### Expected behaviour:

You should see copper ore in the display and the block reader should correctly read a copper ore block

#### Actual behaviour:

The block reader still reads the stone block, after the atomic reconstructor has changed it to copper ore.

#### Steps to reproduce the problem:

1. Set the pictured configuration up. The auto breaker is to the left, the auto placer is to the right, and the atomic reconstructor is behind the target block. Place a lens of the miner on the atomic reconstructor. You'll need to power this (I used a creative mode energy cell).

2. You'll need a lever on the left side of the atomic reconstructor to disable it and the auto breaker. Let the auto placer place a stone block. Activate the lever

3. Place a block reader atop the stone block and a redstone writer top the auto breaker. Cable these together, place a display and a variable store on the cables.

4. Make a constant "stone" block variable (variable A)

5. Make a variable which is the "block" value from the block reader (variable B)

6. Make a new variable "A == B"

7. Place that in the redstone writer atop the auto breaker

8. Make another variable like you did in step 5 and place it in the display.

9. Turn off the lever on the atomic reconstructor.

____
#### Versions:

- This mod: 1.12.2-0.11.3
- CyclopsCore:  1.12.2-0.11.4
- Minecraft:  1.12.2
- Forge: 14.23.1.1601

#### Log file:

No relevant log messages found

![bug](https://user-images.githubusercontent.com/2637504/36130241-7fac2008-1020-11e8-8e98-34503873772e.png)

**Comment by josephcsible** [comment]
Is the Atomic Reconstructor causing a block update? Try putting an Observer against it and see if it fires.

**Comment by RNADNA47** [comment]
Ok. :) I put an observer against the copper ore/stone block space, and can verify it lights up when the reconstructor changes the block from stone to some ore.

**Comment by RNADNA47** [comment]
One more datapoint.

In my creative world, if I place a redstone clock in front of the block reader, everything starts to work fine.

The machine in my survival world is more complex because I used pistons/lava/water and not a stoneworks. In particular, there are two added redstone writers to drive the pistons. When I placed a redstone clock in front of the block reader on that machine, only half the machine worked.
